File: bank-service/src/main/java/rs/edu/raf/model/Currency.java
package rs.edu.raf.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
public class Currency {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true)
    private String name;
    private String code;
    private String symbol;
    @ManyToOne
    private Country country;
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/model/Currency.kt
package org.example.bankservicekotlin.model

import jakarta.persistence.*

@Entity
class Currency(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long?,
    @Column(unique = true)
    val name: String,
    val code: String,
    val symbol: String,
    @ManyToOne
    val country: Country
) {
}